ethstats/ethstats.go

```diff
@@ -251,16 +268,15 @@ func (s *Service) loop() {
 					// proxied validator should sign
 					channel = signCh
 				}
-				// TODO: This is a hacky measure to avoid blocking here if the
-				// channel is not consuming
+				// Avoid blocking here if the channel is not consuming
 				select {
 				case channel <- &statsPayload:
 				default:
 				}
-
+			case <-quitCh:
+				return
```

Comment:
I think this works fine but the indentation is hard to follow - have you tested that it quits cleanly?

Reply:
I haven't tested this yet (I wanted to get the reviews going), but I can do that!

Reply:
Decided to remove this 👍